src/N98/Magento/Command/ScriptCommand.php

```diff
@@ -277,10 +277,11 @@ protected function registerVariable(InputInterface $input, OutputInterface $outp
      */
     protected function runMagerunCommand(InputInterface $input, OutputInterface $output, $commandString)
     {
+        $noInteraction = ! $input->getOption('no-interaction');
         $this->getApplication()->setAutoExit(false);
         $commandString = $this->_replaceScriptVars($commandString);
         $input = new StringInput($commandString);
-        $exitCode = $this->getApplication()->run($input, $output);
+        $input->setInteractive($noInteraction);
```

Comment:
Maybe this could work:

```php
$input = new StringInput($commandString);
$input->setInteractive($noInteraction);
$exitCode = $this->getApplication()->run($input, $output);
```